madgraph/core/diagram_generation.py

```diff
@@ -1730,11 +1730,25 @@ def generate_multi_amplitudes(cls,process_definition,
         except KeyError:
             fstags = []
 
+        try:
+            istags = [leg['is_tagged'] for leg in process_definition['legs'] \
+                 if 'is_tagged' in leg.keys() and leg['state'] == False]
+
+        except KeyError:
+            istags = []
+
         # Generate all combinations for the initial state
         for prod in itertools.product(*isids):
+            if any(istags):
+                if not all(istags):
+                    logger.error("Tagging only one initial-state particle is not allowed")
+                    exit()
```

Comment:
exit quit the code inconditionally which is not nice in interactive mode (and bypass the creation of debug file)

```suggestion
                    raise MadGraph5Error("Tagging only one initial-state particle is not allowed")
```